### PR TITLE
Only log when enabled to avoid unnecessary printing on stdout

### PR DESCRIPTION
**Pull request by baijumeswani**
#724 logs when loading the ort lib. The log can create problems if the logging is not enabled (in particular [this line](https://github.com/microsoft/onnxruntime-genai/blob/32378d96cbb7e921e0d986e5e1a56f086c32fe96/src/logging.cpp#L82)) in debug builds. And it also prints to console unnecessary information.

Changing the behavior in this PR.

### CODE REVIEW

File: src/models/env_utils.cpp [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  // If the function succeeds, the return value is the number of characters stored in the buffer pointed to by lpBuffer, not including the terminating null character.
+  // Therefore, If the function succeeds, kBufferSize should be larger than char_count.
+  auto char_count = ::GetEnvironmentVariableA(var_name, buffer.data(), kBufferSize);
+
+  if (kBufferSize > char_count) {
+    buffer.resize(char_count);
+    return buffer;
+  }
+
+  return std::string();
```
Comment (skyline75489) [inline src/models/env_utils.cpp:34]: This can just be `return {}` if you like

File: src/models/onnxruntime_api.h [hunk 2]
```diff
@@ -93,11 +94,14 @@ p_session_->Run(nullptr, input_names, inputs, std::size(inputs), output_names, o
 #define PATH_MAX (4096)
 #endif
 
-#define LOG_DEBUG(...) Generators::Log("debug", __VA_ARGS__)
-#define LOG_INFO(...) Generators::Log("info", __VA_ARGS__)
-#define LOG_WARN(...) Generators::Log("warning", __VA_ARGS__)
-#define LOG_ERROR(...) Generators::Log("error", __VA_ARGS__)
-#define LOG_FATAL(...) Generators::Log("fatal", __VA_ARGS__)
+#define LOG_WHEN_ENABLED(LOG_FUNC) \
```
Comment (skyline75489) [inline src/models/onnxruntime_api.h:97]: How about move all these macros to `logging.h`, including the androids and other ones the I added in my PRs. That would make this file less bloated.

File: src/models/onnxruntime_api.h [hunk 3]
```diff
@@ -175,6 +179,12 @@ inline void InitApi() {
     return;
   }
 
+  const std::string ort_lib = Generators::GetEnvironmentVariable("ORTGENAI_LOG_ORT_LIB");
+  if (ort_lib == "1" || ort_lib == "true") {
```
Comment (skyline75489) [inline src/models/onnxruntime_api.h:183]: Do we have similar checks elsewhere in the project? Maybe we need a consolidated way to check for `true` and `false`.

Comment (baijumeswani) [reply src/models/onnxruntime_api.h:183]: Will make a utility for this.

File: src/models/onnxruntime_api.h (hunk 2 above)
Comment (baijumeswani) [reply src/models/onnxruntime_api.h:97]: Let's avoid using macros as a general-purpose logging utility for now. If it comes to that, we can make this happen later.

Comment (skyline75489) [reply src/models/onnxruntime_api.h:97]: sure, no problem

**Review by skyline75489 — APPROVED** [review]
Thanks for amending it :)

**Pull request merged**